### PR TITLE
Initial implementation

### PR DESCRIPTION
**Pull request by rmg**
Exposes a single function. Given a path to a directory it returns tar stream. Pipe that tar stream to a file and you have a `.tar` file. Pipe it through `zlib.GZip()` and then to a file and you can call it a `.tgz`.

Connect to strongloop-internal/scrum-nodeops#1015

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+strong-pack
+===========
+
+Simple wrapper for `tar` for packaging a node application without ignoring
+anything except VCS files (`.git`, `.svn`, `.hg`, etc.).
```
Comment (sam-github) [inline README.md]: "without ignoring anything" -> "including everything": I had to read the double negative twice! ;-)

File: README.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+The following example is similar to running `npm pack path/to/app` if npm
+ignored all of the normal rules for ignoring/including files.
+
+```js
+var fs = require('fs');
+var gz = require('zlib').Gzip();
+var pack = require('strong-pack');
+var tgz = fs.createWriteStream('app.tgz', 'binary');
+pack('path/to/app').pipe(gz).pipe(tgz);
```
Comment (sam-github) [inline README.md]: @pthieu You may prefer to use https://www.npmjs.com/package/pump, it avoids some error-prone error-handling.

File: README.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+```
+
+### License
+
+[Artistic License 2.0](https://opensource.org/licenses/Artistic-2.0)
+
+---
+Copyright &copy; 2015, Ryan Graham
```
Comment (sam-github) [inline README.md]: copyright SL in the LICENSE file, no?

File: index.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    path: folder,
+    type: 'Directory',
+    isDirectory: true,
+    filter: FILTER,
+  });
+
+  return contents
+    .on('error', debug.bind(null, 'err reading folder'))
+    .pipe(tarPack)
+    .on('error', debug.bind(null, 'tar creation error'));
```
Comment (sam-github) [inline index.js]: This makes node think that errors have been "handled", but from the external user's point of view they are still "unhandled". Is that intentional? You could do `if (debug.enabled) contents.on('error', ...)`.

File: test/test-pack.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
+'use strict';
+
+var tar = require('tar');
+var test = require('tap').test;
+var pack = require('../');
+var path = require('path');
```
Comment (sam-github) [inline test/test-pack.js:4]: requires aren't sorted

**Comment by sam-github** [comment]
@pthieu PTAL, you can use this for your pack CLI.

File: README.md (hunk 1 above)
Comment (rmg) [inline README.md]: Ah, I think my original wording was a rephrase that referenced npm's .npmignore behaviour more explicitly. I'll reword.

File: index.js (hunk 4 above)
Comment (rmg) [inline index.js]: Good catch.

File: README.md (hunk 2 above)
Comment (rmg) [inline README.md]: @sam-github what would this be with pump? Would be good to include it as an example.

Comment (sam-github) [inline README.md]: I think its:

```
pump(pack('/path/to/app'), gz, tgz, function(err) {
  console.log('pipe finished, maybe with an error:', err)
})
```

**Comment by rmg** [comment]
@sam-github I think I addressed all of your comments (from GitHub's perspective, at least).

**Comment by rmg** [comment]
@slnode test please

**Comment by pthieu** [comment]
LGTM

**Comment by pthieu** [comment]
@rmg -- can't re-assign back to you; collaborators don't have permission.

**Pull request merged**